app/src/main/java/com/devshish/internship/presentation/ui/lyrics/LyricsViewModel.kt

```diff
@@ -57,4 +63,10 @@ class LyricsViewModel(
             }
         }
     }
+
+    private suspend fun isSongStored() {
+        repository.isSongStored(searchSong).collect { isStored ->
+            _isSongStored.value = isStored
+        }
+    }
```

Comment:
Still not correct - in current case you start observing after each click - but you need to observe in once